tests/KubernetesClient.Tests/KubernetesClientConfigurationTests.cs

```diff
@@ -407,6 +409,75 @@ public void LoadKubeConfigStream()
             AssertConfigEqual(expectedCfg, cfg);
         }
 
+        [Fact]
+        public void LoadKubeConfigFromEnvironmentVariable()
+        {
+            // BuildDefaultConfig assumes UseRelativePaths: true, which isn't
+            // done by any tests.
+            var filePath = Path.GetFullPath("assets/kubeconfig.relative.yml");
+
+            Environment.SetEnvironmentVariable("KUBECONFIG", filePath);
+            var cfg = KubernetesClientConfiguration.BuildDefaultConfig();
+
+            Assert.NotNull(cfg);
+        }
+
+        [Fact]
+        public void LoadK8ConfigFromEnvironmentVariable()
+        {
+            var txt = File.ReadAllText("assets/kubeconfig.yml");
+            var expectedCfg = Yaml.LoadFromString<K8SConfiguration>(txt);
+
+            var filePath = Path.GetFullPath("assets/kubeconfig.yml");
+            Environment.SetEnvironmentVariable("KUBECONFIG", filePath);
```

Comment:
This seems bad for parallel execution. These tests need to be part of a collection that don't run in parallel.

Reply:
Yeah, I have two ways I could fix it, but both are based on what public API we want.

If we decided to keep the API where we we can pass in an environment variable name, I can make it so each test uses a unique variable name. 

Otherwise, I'll probably do some InternalsVisibleTo stuff to change the return value for the environment variable.